verif/core_rrv/tests/bubblesort_lfsr.c

```diff
@@ -0,0 +1,89 @@
+
+//------------------------------------------------------------
+// Title : alive_lfsr
+// Project : core_rrv
+//------------------------------------------------------------
+// Description :
+//generating LENGTH pseudo random numbers created using LFSR
+//------------------------------------------------------------
+#include "interrupt_handler.h"
+
+#define LENGTH 15
+
+// Function to generate the next LFSR value
+unsigned int lfsr(unsigned int current_state) {
+    // Feedback polynomial: x^8 + x^6 + x^5 + x^4 + 1
+    // We use taps at positions 8, 6, 5, and 4.
+    unsigned int bit = ((current_state >> 7) ^ (current_state >> 5) ^ (current_state >> 4) ^ (current_state >> 3)) & 1;
+    return (current_state << 1) | bit; // Shift left and add feedback bit
+}
+
+// Function to generate a pseudo-random number in the range 0-99
+unsigned int lfsr_random_0_99(unsigned int seed) {
+    unsigned int random_value = 0;
+    for (int i = 0; i < 7; ++i) { // Generate 7 bits
+        seed = lfsr(seed); // Update LFSR state
+        random_value = (random_value << 1) | (seed & 1); // Append LSB of LFSR state to random_value
+    }
+    return random_value % 100; // Modulate by 100 to get a range of 0-99
+}
+
+void bubbleSort(int arr[], int n) {
+    int i, j;
+    for (i = 0; i < n-1; i++) {
+        for (j = 0; j < n-i-1; j++) {
+            if (arr[j] > arr[j+1]) {
+                int temp = arr[j];
+                arr[j] = arr[j+1];
+                arr[j+1] = temp;
+            }
+        }
+    }
+}
+
+void printSorted(int arr[], int n){
+    int i;
+    for (i = 0; i < n; i++) {
+        rvc_print_int(arr[i]);
+        rvc_printf(" ");
+    }
+}
+
+int check_sort(int arr[], int n){
+    int i;
+    for(i=0; i < n-1; i++){
+        if(arr[i] > arr[i+1])
+            return 0;
+        i++;
+    }
+
+    return 1; 
+}
+int main() {    
+    //enable the mie timer interrupts CSR:
+    unsigned int csr_mie = read_mie();
+    csr_mie = csr_mie | 0x00000888;
+    write_mie(csr_mie); // enable msie, mtie, meie bits in mie
+
+    unsigned int seed = 0xACE1u; // Non-zero seed
+    unsigned int random_numbers[LENGTH];
+
+    // Fill the array with pseudo-random numbers within the range 0-99
+    for (int i = 0; i < LENGTH; ++i) 
+        random_numbers[i] = lfsr_random_0_99(seed + i); // Use varying seeds for each number
```

Comment:
AFAIK, this is not how you should use a SW LFSR random number generator..
The idea is that you call the random number generator and get a value without feeding a seed everytime..

So the better way is to have a global int that holds the current random number, and every time the lfsr function is called it first creates the new random value (and saves it) and then it returns it to whom called that function.